api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
So if I want to use an interface I would need to pass the port in `--listen-addr` and pass the interface in `--listen-interface`, correct? (assuming `listen-addr` has a default port then I only need to pass `--listen-interface`).

If I want to use `--listen-interface` with a different port then I'd have to pass a "bogus" `--listen-addr`.

I think this is acceptable.

Alternative: `--listen-addr eth0:4242`?